bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,371 @@
+import static org.openhab.binding.intesis.internal.api.IntesisBoxMessage.*;
+import static org.openhab.core.thing.Thing.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.intesis.internal.IntesisConfiguration;
+import org.openhab.binding.intesis.internal.IntesisDynamicStateDescriptionProvider;
+import org.openhab.binding.intesis.internal.api.IntesisBoxChangeListener;
+import org.openhab.binding.intesis.internal.api.IntesisBoxIdentity;
+import org.openhab.binding.intesis.internal.api.IntesisBoxMessage;
+import org.openhab.binding.intesis.internal.api.IntesisBoxSocketApi;
+import org.openhab.binding.intesis.internal.enums.IntesisBoxFunctionEnum;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link IntesisBoxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Cody Cutrer - Initial contribution
+ * @author Rocky Amatulli - additions to include id message handling, dynamic channel options based on limits.
+ * @author Hans-Jörg Merk - refactored for openHAB 3.0 compatibility
+ *
+ */
+@NonNullByDefault
+public class IntesisBoxHandler extends BaseThingHandler implements IntesisBoxChangeListener {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisBoxHandler.class);
+    private @Nullable IntesisBoxSocketApi intesisBoxSocketApi;
+
+    private final Map<String, String> properties = new HashMap<>();
+    private final Map<String, List<String>> limits = new HashMap<String, List<String>>();
+
+    private final IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
+
+    private IntesisConfiguration config = new IntesisConfiguration();
+
+    private double minTemp = 0.0d, maxTemp = 0.0d;
```

Comment:
Decimal numbers with point are `double` by default.
```suggestion
    private double minTemp = 0.0, maxTemp = 0.0;
```

Reply:
Done